tools/tertiary-analysis/scanpy/anndata_operations.xml

```diff
@@ -34,6 +40,30 @@ else:
     logging.warning('No genes starting with {} found, skip calculating expression of {} genes'.format('${flag.startswith}', '${flag.flag}'))
 #end for
 
+#if len($copy_o.obs_keys) > 0:
+#for $i, $obs_s in enumerate($copy_o.obs_sources):
+ad_s = sc.read('obs_source_${i}.h5')
+  #for $j, $o_key in enumerate($copy_o.obs_keys):
+keys_to_copy = (k for k in ad_s.obs.keys() if "${o_key.contains}" in k)
+for k_to_copy in keys_to_copy:
+  adata.obs[[k_to_copy+"_${i}"]] = ad_s.obs[[k_to_copy]]
+  if k_to_copy in ad_s.uns.keys():
```

Comment:
@nh3 equivalent to this that I'm doing to copy the metadata of the clustering methods, is there something else I should copy for the obsm dataframes below in terms of metadata (for instance, whatever stores the persplexity used for the tsne). Thanks!

Reply:
This looks good. Checking here is sufficient. anndata ensures consistent dimension and order across the object.